wfapi/wfapi.ipldsch

```diff
@@ -430,4 +430,27 @@ type CatalogMirrorsByPacktype {Packtype:WarehouseAddrList}
 
 type ReplayCapsule union {
 	| Plot "plot.v1"
-} representation keyed
\ No newline at end of file
+} representation keyed
+
+# MirroringConfig defines the mirroring configuration for an entire workspace.
+# This maps the addresses which we want to publish to (of type WarehouseAddr) 
+# to specific configs, which may be backed by various types of storage.
+type MirroringConfig {WarehouseAddr:WarehouseMirroringConfig}
+
+type WarehouseMirroringConfig struct {
+	pushConfig WarehousePushConfig 
+}
+
+type WarehousePushConfig union {
+	| S3PushConfig "s3"
+	| MockPushConfig "mock"
+} representation keyed
+
+type S3PushConfig struct {
+	endpoint String
+	region String
+	bucket String
+	path optional String
+}
+
+type MockPushConfig struct {}
```

Comment:
For the purpose of getting it visible in this PR: here's copypaste of the (real!) example config from [the related branch in warpsys](https://github.com/warptools/warpsys/blob/a22c19624ea6e8902700e68e9e6cd912693a4d4d/.warpforge/mirroring.json):

```
{
	"ca+https://warpsys-wares.s3.fr-par.scw.cloud": {
		"pushConfig": {
			"s3": {
				"endpoint": "https://s3.fr-par.scw.cloud",
				"region": "fr-par",
				"bucket": "warpsys-wares"
			}
		}
	}
}
```

Reply:
I think we're probably missing a capsule type here.  Something that says what this data is, and gives us a position to put version indicators.

It's possible we might want to think about splitting some of this into another schema file too -- to group together all the types that are only used to describe values that appear only in workspace config.